core/app/api/customer/groups/route.ts

```diff
@@ -0,0 +1,41 @@
+import { NextResponse } from 'next/server';
+
+import { CustomerGroupsSchema } from '~/lib/makeswift/components/customer-group-slot/schema';
+
+export async function GET() {
+  if (!process.env.BIGCOMMERCE_ACCESS_TOKEN) {
+    // eslint-disable-next-line no-console
+    console.error('BIGCOMMERCE_ACCESS_TOKEN is not set');
+
+    return NextResponse.json({ status: 'error', error: 'Internal server error' }, { status: 500 });
+  }
+
+  try {
+    const response = await fetch(
+      `https://api.bigcommerce.com/stores/${process.env.BIGCOMMERCE_STORE_HASH}/v2/customer_groups`,
+      {
+        method: 'GET',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+          'X-Auth-Token': process.env.BIGCOMMERCE_ACCESS_TOKEN,
+        },
+      },
+    );
+
+    const jsonResponse: unknown = await response.json();
+    const result = CustomerGroupsSchema.safeParse(jsonResponse);
+
+    if (!result.success) {
+      throw result.error;
+    }
```

Comment:
If we're just throwing the error like this, then call `parse` instead of `safeParse`